tryhttp.go

```diff
@@ -0,0 +1,120 @@
+// Package tryhttp reschedule failed HTTP requests.
+//
+// TODO: document more.
+package tryhttp
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/teamwork/utils/stringutil"
+)
+
+// Client to perform HTTP requests.
+type Client struct {
+	// Retry determines if we should retry the request after failure. It will
+	// keep retrying until the second argument returns false. The first return
+	// argument can be used to delay the HTTP request.
+	Retry func(r *http.Request, attempt int) (delay time.Duration, retry bool)
```

Comment:
You should specify what we consider a failure.  Is it by http status code?  Something in the body (probably not, but still be good to clarify).